packages/google_sign_in/google_sign_in/example/ios/RunnerTests/GoogleSignInTests.m

```diff
@@ -26,62 +26,406 @@ - (void)setUp {
   [super setUp];
   self.mockBinaryMessenger = OCMProtocolMock(@protocol(FlutterBinaryMessenger));
   self.mockPluginRegistrar = OCMProtocolMock(@protocol(FlutterPluginRegistrar));
-  self.mockSharedInstance = [OCMockObject partialMockForObject:[GIDSignIn sharedInstance]];
+
+  id mockSharedInstance = OCMClassMock([GIDSignIn class]);
+  OCMStub([mockSharedInstance sharedInstance]).andReturn(mockSharedInstance);
```

Comment:
Any chance we could adjust the code to accept an injectable GIDSignIn instance instead of using class mocking?

Reply:
Done.